app/commands/v2/publish.rb

```diff
@@ -27,6 +32,28 @@ def validate!
         ) unless update_type.present?
       end
 
+      def validate_version_lock!
+        current_content_item = DraftContentItem.find_by(content_id: content_id)
+        current_version = Version.find_by(target: current_content_item)
+        previous_version_number = payload[:previous_version]
```

Comment:
Given this business logic is duplicated in the various command objects, would it be possible to move it to the model layer?

```ruby
Version.conflict?(current_content_item, payload[:previous_version]) # Something like this?
```